src/Database/Type/BinaryType.php

```diff
@@ -55,6 +55,9 @@ public function toPHP($value, Driver $driver)
         if ($value === null) {
             return null;
         }
+        if (is_string($value) && preg_match('/^[a-zA-Z0-9]+$/', $value)) {
```

Comment:
Actually testing the type will require loading the Sqlserver driver class, which we can't do as PDO constants are only defined when pdo_sqlsrv is installed.